akka-stream/src/main/scala/akka/stream/Graph.scala

```diff
@@ -67,6 +68,12 @@ trait Graph[+S <: Shape, +M] {
    * less specific than attributes set directly on the individual graphs of the composite.
    */
   def addAttributes(attr: Attributes): Graph[S, M] = withAttributes(traversalBuilder.attributes and attr)
+
+  /**
+   * Transform the materialized value of this Flow, leaving all other properties as they were.
+   */
+  def mapMaterializedValue[Mat2](f: M => Mat2): Graph[S, Mat2] =
```

Comment:
Since the parameter type differs between scaladsl and javadsl but both types are SAMs we won't be able to add this here, there is no way we can make a common method in the supertype that satisfies both and keep source and binary compatibility.

Reply:
can we an an intermediate ScalaGraph trait? not that I'm sure it'd do us any good...